lib/pushmi_pullyu/preservation_queue.rb

```diff
@@ -61,11 +61,11 @@ def next_item
   end
 
   def wait_next_item
-    loop do
-      element = next_item
-      return element if element.present?
-      sleep @poll_interval
-    end
+    # loop do
```

Comment:
Good point. That won't play well with the sleep/poll interval, either. Let me think about this for a bit